### ISSUE TITLE
Type aliases

### ISSUE THREAD

**Issue by EmilyBourne** [pyccel/pyccel#1544]
### Relevant Discussion

https://github.com/pyccel/pyccel/issues/1487#issuecomment-1702207633

### Describe the feature
As suggested by @adam-urbanczyk .

It would be nice to have support for type aliases. This would allow the user to change the types of multiple functions at the same time. E.g. it is common to use this in C++ with a type alias `using Real = double;`

### Test Code
Provide code which does not currently work but which should do when this issue is fixed:
```python
from typing import TypeAlias

MyType: TypeAlias = int
```

### Proposed Solution
This could easily be implemented after #1523 by saving the type alias as a symbolic alias in the scope. An error should be raised if the type cannot be determined at compile time (e.g. because it is defined in an if block).

In order to implement this the following steps should be taken:
1. In `pyccel/ast/typingext.py` create a class inheriting from `pyccel.ast.datatypes.DataType` which describes the type `TypeAlias`
2. Add the module to the `pyccel.ast.utilities.builtin_import_registry`
3. In the semantic stage, make the necessary changes in `_visit_Assign` such that if the `lhs` is a `TypeAlias`, the alias is added to the scope using `pyccel.parser.scope.Scope.insert_symbolic_alias` and an `EmptyNode()` is returned
4. Add tests

**Comment by adesh1998** [comment]
is this issue still open if so could you assign this issue to me?

**Comment by EmilyBourne** [comment]
Hi @adesh1998 , thanks for helping! I've assigned the issue to you. Feel free to contact me on [discord](https://discord.gg/2Q6hwjfFVb) if you need more information/help setting up your development environment/etc

**Comment by EmilyBourne** [comment]
Hi @adesh1998 I just wanted to check in with you. Have you managed to make a start on this? Do you need any help?
If you have not managed to start then are you still planning to work on this?

**Comment by adesh1998** [comment]
You can assign somebody i will be busy for couple of days

**Comment by EmilyBourne** [comment]
Should be tackled after #1683 is merged
